doc/api/buffer.md

```diff
@@ -92,12 +104,12 @@ to one of these new APIs.*
   initialized `Buffer` of the specified size. This method is slower than
   [`Buffer.allocUnsafe(size)`][`Buffer.allocUnsafe()`] but guarantees that newly
   created `Buffer` instances never contain old data that is potentially
-  sensitive.
+  sensitive. A `TypeError` will be thrown if `size` is not a number.
 * [`Buffer.allocUnsafe(size)`][`Buffer.allocUnsafe()`] and
   [`Buffer.allocUnsafeSlow(size)`][`Buffer.allocUnsafeSlow()`] each return a
   new uninitialized `Buffer` of the specified `size`. Because the `Buffer` is
   uninitialized, the allocated segment of memory might contain old data that is
-  potentially sensitive.
+  potentially sensitive. A `TypeError` will be thrown if `size` is not a number.
```

Comment:
We rarely document any error cases and I personally would rather keep them out of the docs to keep the description brief.

Reply:
error cases are documented in buffer.md:
```
% grep "will be thrown" doc/api/buffer.md 
  sensitive. A `TypeError` will be thrown if `size` is not a number.
A `TypeError` will be thrown if `size` is not a number.
A `TypeError` will be thrown if `size` is not a number.
A `TypeError` will be thrown if `size` is not a number.
A `TypeError` will be thrown if `array` is not an `Array` or other type
A `TypeError` will be thrown if `arrayBuffer` is not an [`ArrayBuffer`][] or a
A `TypeError` will be thrown if `buffer` is not a `Buffer` or other type
A `TypeError` will be thrown if `object` has not mentioned methods or is not of
A `TypeError` will be thrown if `string` is not a string or other type
```

But given its documented elsewhere, I removed it from here.